### ISSUE TITLE
Jsx namespaced name and attributes

### ISSUE THREAD

**Issue by armano2** [microsoft/TypeScript#42963]
# Bug Report

### 🔎 Search Terms

- jsx
- ast
- syntax
- JSXNamespacedName
- typescript 4.2

### 🕗 Version & Regression Information

- This changed between versions 4.2.0 and 4.3.x

### ⏯ Playground Link

[Playground link](https://www.typescriptlang.org/play?ts=next#code/CYUwxgNghgTiAEA7KBbEBnADlMCBSAygBrwDeAsAFDzwCWiALiDAGY4ICSjM96tYAUQgg0jdGSo0aAcgYALWugBcAI2lL4URAE8A3JKnSVSsOonUplgB4at2gzQC++izKiqzdl06qOqVAB53FQA+AIB6YJCXQOMwADorMPC4xOj-SgD5RVVk7OVQlyA)
https://github.com/typescript-eslint/typescript-eslint/pull/3112

### 💻 Code

```ts
declare namespace JSX {
  interface IntrinsicElements {
    'this:b': any;
    'b:c': {
        x: any
    };
    'a:b': any;
  }
}

<a:b></a:b>;  // is working
<b:c.x></b:c.x>;  // should fail, there is no syntax error reported, and only issue is related to incorrect type
<this:b></this:b>; // its failing but there is no syntax error reported
```

### 🙁 Actual behavior

Invalid syntax is not reported as such. 

`this:b` in AST is reporter as `ThisExpression`, and has range of `this:b`

when code above is run in preserve mode, this syntax is generated
```ts
<a:b></a:b>;
<b:c.x></b:c.x>;
<this></this>;
```

this syntax is not allowed according to spec
```ts
<b:c.x>
```

```js
JSXMemberExpression :
    JSXIdentifier . JSXIdentifier
    JSXMemberExpression . JSXIdentifier
```
https://github.com/facebook/jsx

### 🙂 Expected behavior

We should receive some sort of error or `this:b` should be parsed as `Identifier`
